src/compiler/config/validate-service-worker.ts

```diff
@@ -17,18 +37,18 @@ export const validateServiceWorker = (config: d.ValidatedConfig, outputTarget: d
     return;
   }
 
-  if (outputTarget.serviceWorker === true) {
-    outputTarget.serviceWorker = {};
-  } else if (!outputTarget.serviceWorker && config.devMode) {
+  if (!outputTarget.serviceWorker && config.devMode) {
     outputTarget.serviceWorker = null;
     return;
   }
 
-  if (typeof outputTarget.serviceWorker !== 'object') {
-    // what was passed in could have been a boolean
-    // in that case let's just turn it into an empty obj so Object.assign doesn't crash
-    outputTarget.serviceWorker = {};
-  }
+  outputTarget.serviceWorker = {
+    ...outputTarget.serviceWorker,
+    globDirectory: outputTarget.serviceWorker?.globDirectory ?? outputTarget.appDir,
```

Comment:
Since we're validating user input here, it's probably best we keep the `typeof` check on `globDirectory` (which we could refactor to `isString()`) that [we originally had](https://github.com/ionic-team/stencil/pull/3813/files#diff-c2b7ab30b2bdcc180082ae16ba7b8aa1da392a53ea5054cea9bfb506b98d3fe0L41). That way we don't allow something like a numeric literal or object literal to be used for the value of `globDirectory`